js-practive/js/main.js

```diff
@@ -1 +1,260 @@
+        })
+        .then((res)=>{
+            if (res.ok){
+                renderUserHTM(user)
+                clearForm()
+                clearHTML()
+                getUsers()
+            }
+        });
+    } else {
+        fetch('/user', {
+            method: 'POST',
+            body: JSON.stringify(user),
+        })
+        .then((res)=>{
+            if (res.ok){
+                renderUserHTM(user)
+                clearForm()
+            }
+        });
+    }
+    deleteDataOnLocalStorage()
+    goToTable()
+}
+
+function getUserFromForm() {
+
+    setHtmlElementUserForm()
+
+    return {
+        id: userForm.elements['id'].value,
+        fullName: userForm.elements['fullname'].value,
+		birthday: userForm.elements['birthday'].value,
+		profession: userForm.elements['profession'].value,
+		address: userForm.elements['address'].value,
+		country: userForm.elements['country'].value,
+		shortInfo: userForm.elements['short-info'].value,
+		fullInfo: userForm.elements['full-info'].value, 
+    }
+}
+
+function renderUserHTM(user) {
+    const userValueRenderHtml = generateUserHTML(user);
+
+    usersTable.insertAdjacentHTML("beforeend", userValueRenderHtml);
+}
+
+function onUsersTableClick(e) {
+    const user = getUserBySelector(e.target);
+    const id = user.dataset.id;
+
+    if (user) {
+        if (e.target.classList.contains('editBtn')) {
+            goToUserForm(id)
+            getSingleUser(id)
+                .then((user) => {
+                    setUserData(user)
+                })
+        }
+        if (e.target.classList.contains('deleteBtn')) {
+            deleteUser(id)
+                .then(() => {
+                    clearHTML()   
+                })
+                .then(() => {
+                    getUsers();
+                })
+        }
+    }
+}
+
+function getUserBySelector(el) {
+    return el.closest('.user');
+}
+
+
+function setUserData(user){
+    setHtmlElementUserForm()
+
+    userForm.elements['id'].value = user.id
+    userForm.elements['fullname'].value = user.fullName
+    userForm.elements['birthday'].value = user.birthday
+    userForm.elements['profession'].value = user.profession
+    userForm.elements['address'].value = user.address
+    userForm.elements['short-info'].value = user.shortInfo
+    userForm.elements['full-info'].value = user.fullInfo
+}
+
+function getSingleUser(userId) {
+    return fetch(`/user?id=${userId}`, {
+        method: 'GET',
+    })
+    .then(res => {
+        return res.json();
+    })
+}
+
+function deleteUser(userId) {
+    return fetch(`/user?id=${userId}` , {
+        method: 'DELETE',
+    })
+    .then(res => {
+        return res.ok;
+    });
+}
+
+
+function clearForm() {
+    userForm.reset()     
+}
+
+function clearHTML() {
+    let tableRows = usersTable.getElementsByTagName('tr');
+    do {
+        usersTable.removeChild(tableRows[0])
+    } while (tableRows[0])
+}
+
+function setDataOnLocalStorage(user) { 
+    if(!user.id){
```

Comment:
you can do this operation in place where you use this function. Here just set user. 
Cause in case of such implementation you make your function dependant on buissness logic.